core/src/utilities/helpers/routing-helpers.js

```diff
@@ -264,6 +264,130 @@ class RoutingHelpersClass {
       featureToggleList.forEach(ft => LuigiFeatureToggles.setFeatureToggle(ft));
     }
   }
+
+  /**
+   * This function takes an intentLink and parses it conforming certain limitations in characters usage.
+   * Limitations include:
+   *  - `semanticObject` allows only alphanumeric characters
+   *  - `action` allows alphanumeric characters and the '_' sign
+   *
+   * Example of resulting output:
+   * ```
+   *  {
+   *    semanticObject: "Sales",
+   *    action: "order",
+   *    params: [{param1: "value1"},{param2: "value2"}]
+   *  };
+   * ```
+   * @param {string} link  the intent link represents the semantic intent defined by the user
+   *                        i.e.: #?Intent=semanticObject-action?param=value
+   */
+  getIntentObject(intentLink) {
+    const intentParams = intentLink.split('?Intent=')[1];
+    if (intentParams) {
+      const elements = intentParams.split('-');
+      if (elements.length === 2) {
+        // avoids usage of '-' in semantic object and action
+        let semanticObject = elements[0];
+        let actionAndParams = elements[1].split('?');
+        // length 2 involves parameters, length 1 involves no parameters
+        if (actionAndParams.length === 2 || actionAndParams.length === 1) {
+          let action = actionAndParams[0];
+          let params = actionAndParams[1];
+          // parse parameters, if any
+          if (params) {
+            params = params.split('&');
+            let paramObject = [];
+            params.forEach(item => {
+              const param = item.split('=');
+              param.length === 2 && paramObject.push({ [param[0]]: param[1] });
+            });
+            params = paramObject;
+          }
+          const alphanumeric = /^[0-9a-zA-Z]+$/;
+          const alphanumericOrUnderscore = /^[0-9a-zA-Z_]+$/;
+          // TODO: check for character size limit
+          if (
+            semanticObject.match(alphanumeric) &&
+            action.match(alphanumericOrUnderscore)
+          ) {
+            return {
+              semanticObject: semanticObject,
+              action: action,
+              params: params
+            };
```

Comment:
ES6 removes all of that repetition
```suggestion
            return {
              semanticObject,
              action,
              params
            };
```